core/src/main/java/com/linecorp/armeria/common/util/EventLoopThreadFactory.java

```diff
@@ -16,31 +16,30 @@
 
 package com.linecorp.armeria.common.util;
 
-import static java.util.Objects.requireNonNull;
-
 import java.util.concurrent.ThreadFactory;
+import java.util.function.Function;
 
 import javax.annotation.Nullable;
 
-import io.netty.util.concurrent.DefaultThreadFactory;
-
 /**
  * {@link ThreadFactory} that creates event loop threads.
  *
  * @see EventLoopGroups
+ *
+ * @deprecated Use {@link ThreadFactories#newEventLoopThreadFactory(String, boolean)} or
+ * {@link ThreadFactories#builder(String)}. Note that setting
+ * {@link ThreadFactoryBuilder#eventLoop(boolean)} true is required to create event loop threads, e.g.
+ * {@code ThreadFactories.builder("myThreadNamePrefix").eventLoop(true).build()}.
  */
-public final class EventLoopThreadFactory implements ThreadFactory {
-
-    // Note that we did not extend DefaultThreadFactory directly to hide it from the class hierarchy.
-    private final ThreadFactory delegate;
-
+@Deprecated
+public final class EventLoopThreadFactory extends AbstractThreadFactory {
     /**
      * Creates a new factory that creates a non-daemon and normal-priority thread.
      *
      * @param threadNamePrefix the prefix of the names of the threads created by this factory.
      */
     public EventLoopThreadFactory(String threadNamePrefix) {
-        this(new EventLoopThreadFactoryImpl(requireNonNull(threadNamePrefix, "threadNamePrefix")));
+        super(threadNamePrefix, false, Thread.NORM_PRIORITY, null, Function.identity());
```

Comment:
I think we can just call another constructor so that we do not have to specify
default values so many times:
```java
public EventLoopThreadFactory(String threadNamePrefix) {
    this(threadNamePrefix, false);
}

public EventLoopThreadFactory(String threadNamePrefix, boolean daemon) {
    this(threadNamePrefix, daemon, Thread.NORM_PRIORITY);
}

public EventLoopThreadFactory(String threadNamePrefix, int priority) {
    this(threadNamePrefix, false, priority);
}

...
```
In that way, we can just change the default value once if we need to. 😉

Reply:
Changed it. Thank you.